opennlp-models-test/src/main/java/org.apache.opennlp/Main.java

```diff
@@ -0,0 +1,106 @@
+ * limitations under the License.
+ */
+package org.apache.opennlp;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.jar.JarFile;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+
+public class Main {
+
+  public static void main(String[] args) {
+    if (args.length != 1) {
+      throw new IllegalArgumentException("This tool expects at least one argument");
+    }
+    System.err.println("Executing basic model validation checks.");
+
+    final Path testBaseDir = Path.of(args[0]);
+    final Path projectDir = testBaseDir.getParent();
+    final List<String> expectedModels = getExpectedModels();
+
+    final String pattern = "opennlp-models.*\\.jar";
+
+    final List<Path> availableModelJars = getAvailableModelJars(pattern, testBaseDir, projectDir);
+
+    if (expectedModels.size() != availableModelJars.size()) {
+      throw new IllegalArgumentException("Detected a mismatch between " +
+          "expected and available models! " +
+          "Expected: " + expectedModels.size() +
+          "; Actual: " + availableModelJars.size());
+    }
+
+    for (String model : expectedModels) {
+      boolean found;
+      for (Path availableJar : availableModelJars) {
+        found = isModelInJar(availableJar, model);
+        if (found) {
+          return;
+        }
+      }
+      throw new IllegalArgumentException(
+          "Expected model '" + model + "' could not be found inside the generated JAR files!");
+    }
+
+
+  }
+
+  public static boolean isModelInJar(Path jarFilePath, String expectedModel) {
+    try (JarFile jarFile = new JarFile(jarFilePath.toFile())) {
+      return jarFile.stream()
+          .anyMatch(entry -> entry.getName().equals(expectedModel));
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to read the JAR file: " + jarFilePath, e);
+    }
+  }
+
+  private static List<Path> getAvailableModelJars(String pattern, Path testDir, Path projectDir) {
+    final Pattern regexPattern = Pattern.compile(pattern);
+    try (Stream<Path> stream = Files.walk(projectDir)) {
+      return stream
+          .filter(Files::isRegularFile)
+          .filter(path -> !path.startsWith(testDir))
+          .filter(path -> regexPattern.matcher(path.getFileName().toString()).matches())
+          .toList();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static List<String> getExpectedModels() {
+    try (InputStream inputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream("expected-models.txt")) {
+      if (inputStream == null) {
+        throw new IllegalArgumentException("Expected model file could not be found!");
+      }
+
+      try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
```

Comment:
This can be defined after line 92 in the first try-with-res.

Reply:
If it is done there, it can result in a NPE.